examples/custom_shape/stages.py

```diff
@@ -0,0 +1,59 @@
+from locust import HttpUser, TaskSet, task, constant
+from locust import LoadTestShape
+
+
+class UserTasks(TaskSet):
+    @task
+    def get_root(self):
+        self.client.get("/")
+
+
+class WebsiteUser(HttpUser):
+    wait_time = constant(0.5)
+    tasks = [UserTasks]
+
+
+class StagesShape(LoadTestShape):
+    """
+    A simply load test shape class that has different user and hatch_rate at
+    different stages.
+
+    Keyword arguments:
+
+        stages -- A list of dicts, each representing a stage with the following keys:
+            duration -- When this many seconds pass the test is advanced to the next stage
+            users -- Total user count
+            hatch_rate -- Hatch rate
+            stop -- A boolean that can stop that test at a specific stage
+
+        stop_at_end -- Can be set to stop once all stages have run.
+    """
+    def __init__(self,
+            stages=[
+                {'duration': 60, 'users': 10, 'hatch_rate': 10},
+                {'duration': 80, 'users': 30, 'hatch_rate': 10},
+                {'duration': 100, 'users': 50, 'hatch_rate': 10},
+                {'duration': 120, 'users': 70, 'hatch_rate': 10},
+                {'duration': 180, 'users': 100, 'hatch_rate': 10},
+                {'duration': 220, 'users': 10, 'hatch_rate': 10},
+                {'duration': 230, 'users': 5, 'hatch_rate': 10},
```

Comment:
Include a stage with ramp down just to show that it can be done :)

Reply:
This does ramp down, see `users` key, it goes up and then down 🙂

But OK, I'll reduce number of stages here.

Reply:
Ah, my comment was about having so many stages, not about the very last stage :)